test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env rspec
+
+root_path = File.expand_path('../..', __FILE__)
+ENV["Y2DIR"] = File.join(root_path, 'src')
+
+require "yast"
+require_relative "../../../src/include/autoinstall/ask.rb"
+require_relative "../../../src/modules/Profile"
+
+describe "Yast::AutoinstallAskInclude" do
+  module DummyYast
+    class AutoinstallAskClient < Yast::Client
+      def main
+        Yast.include self, "autoinstall/ask.rb"
+      end
+
+      def initialize
+        main
+      end
+    end
+  end
+
+  let(:client) { DummyYast::AutoinstallAskClient.new }
+  let(:base_ask) do
+    { "type" => "string", "question" => "hostname?", "default" => "my.site.de",
+      "help" => "Some help", "dialog" => 0, "element" => 0,
+      "stage" => "initial" }
+  end
+  let(:profile) { { "general" => { "ask-list" => ask_list } } }
+  let(:result) { :ok }
+
+  describe "#askDialog" do
+    let(:ask_list) { [ask] }
+
+    before(:each) do
+      allow(Yast::Profile).to receive(:current).and_return(profile)
+      allow(Yast::Stage).to receive(:initial).and_return(true)
+      allow(Yast::UI).to receive(:UserInput).and_return(result)
+    end
+
+    context "when ask-list is empty" do
+      let(:ask_list) { [] }
+
+      it "no dialog is shown" do
+        expect(Yast::UI).to_not receive(:OpenDialog)
+        client.askDialog
+      end
+    end
+
+    describe "dialogs creation" do
+      context "when the ask-list contains a question with type 'string'" do
+        let(:ask) { base_ask }
+
+        it "creates a TextEntry widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
```

Comment:
This is a valid point. But I preferred to keep calling original methods just because askDialog is a huge method and outcomes of those calls could be used later. Otherwise, I would to write a lot of code to mock things.